frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions, searchSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from './search';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean,
+}
+interface Props {
+  submissions: any,
+  dispatch: any,
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // eslint-disable-next-line react/no-unused-state
+      loading: props.submissions.submissions === undefined,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submission === undefined) {
+      this.getData();
+    }
+  }
+
+  getData = async () => {
+    const { submissions, dispatch } = this.props;
+
+    try {
+      const response = await ExerciseService.getSubmissions(submissions.page, '');
+      const jsonResp = await response.json();
```

Comment:
This `.json` should be made in the service. Otherwise we'll need to call it every time.

Reply:
I see you've marked this comment as resolved but the `.json` is still here 🤔